### PR TITLE
Speed up tests

### LINKED ISSUE #51: Speed up Tests
Currently the tests take 42 minutes to run (total). A lot of that is in adding and removing of packages/updating manifests.

TODO: run some benchmarks and see where the time is taken and then reduce

### PR DESCRIPTION
**Pull request by mmiller-max**
Not sure how much this will affect test speed, so a WIP PR.

Will close #51 eventually.

### CODE REVIEW

**Comment by mmiller-max** [comment]
OK this is better, but still not great. On Ubuntu and Julia V1 its gone from 8m to 5m30s. Changes were:

- Don't activate test package environment
- Stop checking registry as much
- Remove tests that duplicate testing
- Simplify all test packages
- Don't clone a fresh registry - we're not testing it

I think most of our time is in the creation of a new Julia process each time `TestReports.test` is executed. So reducing the time further might be possible by combining tests. However I think it's fine for now, it's not a huge concern and at least it's better.

Open to any suggestions, but also happy just to close this and move on. Can always revisit it later.

File: test/reportgeneration.jl [hunk 1]
```diff
@@ -17,23 +17,23 @@ end
 @testset "Runner tests" begin
     @testset "Installed packages by name in primary environment" begin
         # Pkg.add
+        Pkg.UPDATED_REGISTRY_THIS_SESSION[] = true
```
Comment (oxinabox) [inline test/reportgeneration.jl:20]:
```suggestion
        # for speed avoid updating registry, we have a fresh one anyway
        Pkg.UPDATED_REGISTRY_THIS_SESSION[] = true```

File: test/test_packages/CustomTestSet/test/runtests.jl [hunk 2]
```diff
@@ -42,29 +42,4 @@ end
         @test true
     end
     @test true
-end
-
-# Own time_taken field
-mutable struct TimeTakenTestSet <: AbstractTestSet
```
Comment (oxinabox) [inline test/test_packages/CustomTestSet/test/runtests.jl:48]: I see this have been move to not be in a seperate process, that makes sense

**Review by oxinabox — APPROVED** [review]
LGTM

**Pull request merged**